### PR TITLE
Webrobot on EC2 with fabric

### PR DESCRIPTION
**Pull request by eranezer**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -270,16 +275,25 @@ def _create_firefox():
...
-    driver = webdriver.Firefox(options=options, desired_capabilities=cap)
+    driver = webdriver.Firefox(service=FirefoxService(GeckoDriverManager().install()), options=options,
+                               desired_capabilities=cap)
     return driver
 
 
+@app.route('/health_check', methods=['get'])
+def health_check():
+    return{
+        "DISPLAY": os.getenv('DISPLAY'),
```
Comment (rdar-lab) [inline app.py]: This may be a security risk, please remove it

File: app.py [hunk 2]
```diff
@@ -270,16 +275,25 @@ def _create_firefox():
...
+        "PATH": os.getenv('PATH')
```
Comment (rdar-lab) [inline app.py]: Save as above

File: deploy/installation/web_robot_insatall.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        put(local_path=self.local_app_dir + service_file, remote_path="/etc/systemd/system/" + service_file,
+            use_sudo=True)
+        sudo("systemctl daemon-reload")
+        sudo("systemctl start vncserver@:1.service")
+        sudo("systemctl enable vncserver@:1.service")
+
+    def set_display_param(self):
+        run("echo export DISPLAY=:1 >> ~/.bashrc && source ~/.bashrc")
+        run("echo export DISPLAY=:1 >> ~/.profile")
+        # sudo("echo export DISPLAY=:1 >> /etc/environment")
```
Comment (rdar-lab) [inline deploy/installation/web_robot_insatall.py]: Remove unnecessery comments

File: deploy/installation/web_robot_insatall.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        self.install_and_initiate_nginx()
+
+    def apt_update_and_upgrade(self):
+        helpers.print_title("Update and upgrade to make sure the system running the latest packages")
+        sudo("apt update")
+        sudo("apt -y upgrade")
+
+    def install_gui_pagckages(self):
+        helpers.print_title("Install all the GUI packages")
+        # sudo("apt -y install python3-pip python3-dev build-essential libssl-dev libffi-dev python3-setuptools")
```
Comment (rdar-lab) [inline deploy/installation/web_robot_insatall.py]: Remove unnecessary comments

File: deploy/installation/web_robot_insatall.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        # sudo("echo export DISPLAY=:1 >> /etc/environment")
+
+    def deploy_app(self):
+        if files.exists('/home/ubuntu/web-robot-app'):
+            helpers.print_title("Delete old app files")
+            run('rm -rf /home/ubuntu/web-robot-app')
+
+        helpers.print_title("Copy app files")
+        run("mkdir web-robot-app")
+        # app_files = os.listdir(app_dir)
```
Comment (rdar-lab) [inline deploy/installation/web_robot_insatall.py]: Remove unnecessary comments

File: deploy/installation/web_robot_insatall.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+    def install_virtualenv_and_requirements(self):
+        run("sudo apt -y install python3-pip")
+        run("pip install virtualenv")
+        with cd("/home/ubuntu/web-robot-app"):
+            run("python3 -m virtualenv venv")
+            with prefix('source venv/bin/activate'):
+                run('pip install -r requirements.txt')
+                run('deactivate')
+            # run("venv/bin/pip install -r ./requirements.txt")
```
Comment (rdar-lab) [inline deploy/installation/web_robot_insatall.py]: Remove unnecessary comments

File: deploy/installation/web_robot_insatall.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    def restart_nginx(self):
+        sudo("systemctl restart nginx")
+
+    def install_google_chrome(self):
+        helpers.print_title("Install google-chrome")
+        run("wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb")
+        sudo("dpkg -i google-chrome-stable_current_amd64.deb")
+        run("rm google-chrome-stable_current_amd64.deb")
+
+    # def install_chromium_browser(self):
```
Comment (rdar-lab) [inline deploy/installation/web_robot_insatall.py]: Remove unnecessary comments

File: vncserver@:1.service [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+Description=Remote desktop service (VNC)
+After=syslog.target network.target
+
+[Service]
+Type=forking
+User=ubuntu
+
+# Clean any existing files in /tmp/.X11-unix environment
+ExecStartPre=/bin/sh -c '/usr/bin/vncserver -kill %i > /dev/null 2>&1 || :'
+#ExecStart=/usr/bin/vncserver -geometry 1800x1000 -depth 16 -dpi 120 -alwaysshared -localhost %i
```
Comment (rdar-lab) [inline vncserver@:1.service]: Remove unnecessary comments

File: deploy/installation/fabfile.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+@task
+def deploy_app():
+    """
+    Deploy new version
+    """
+    web_robot_tasks.deploy_app()
+
+
+# Run the tasks with the following command:
```
Comment (rdar-lab) [inline deploy/installation/fabfile.py]: Remove unnecessary comments

**Review by rdar-lab — CHANGES_REQUESTED** [review]

File: app.py (hunk 1 above)
Comment (eranezer) [reply app.py]: Do you want me to return a string instead the parameters like "ok", or remove the endpoint?

**Pull request merged**